src/import-export/csv-imp/gnc-tx-import.cpp

```diff
@@ -0,0 +1,960 @@
+    for (orig_lines_it, odd_line;
+            orig_lines_it != orig_lines_max;
+            ++orig_lines_it, odd_line = !odd_line)
+    {
+        prop_map_t trans_props;
+
+        /* Skip current line if:
+           1. only looking for lines with error AND no error on current line
+           OR
+           2. looking for all lines AND
+              skip_rows is enabled AND
+              current line is an odd line */
+        if ((redo_errors && orig_lines_it->second.empty()) ||
+           (!redo_errors && skip_rows && odd_line))
+            continue;
+
+        auto line = orig_lines_it->first;
+        GncCsvTransLine* trans_line = NULL;
+
+        /* Convert this import line into a map of transaction/split properties. */
+        auto loop_err = false;
+        auto col_types_it = column_types.cbegin();
+        auto line_it = line.cbegin();
+        for (col_types_it, line_it;
+                col_types_it != column_types.cend(),
+                line_it != line.cend();
+                ++col_types_it, ++line_it)
+        {
+            std::shared_ptr<GncTransProperty> property;
+            switch (*col_types_it)
+            {
+                case GncTransPropType::DATE:
+                    property = GncTransPropImpl<time64*>::make_new (*line_it, date_format);
+                break;
+
+                case GncTransPropType::DESCRIPTION:
+                case GncTransPropType::NOTES:
+                case GncTransPropType::MEMO:
+                case GncTransPropType::OMEMO:
+                case GncTransPropType::NUM:
+                    property = GncTransPropImpl<std::string*>::make_new (*line_it);
+                    break;
+
+                case GncTransPropType::ACCOUNT:
+                case GncTransPropType::OACCOUNT:
+                    property = GncTransPropImpl<Account*>::make_new (*line_it);
+                    if (*col_types_it == GncTransPropType::ACCOUNT)
+                        home_account = dynamic_cast<GncTransPropImpl<Account*>*>(property.get())->value;
+                    break;
+
+                case GncTransPropType::BALANCE:
+                case GncTransPropType::DEPOSIT:
+                case GncTransPropType::WITHDRAWAL:
+                    property = GncTransPropImpl<gnc_numeric*>::make_new (*line_it, currency_format);
+                    break;
+
+                default:
+                    continue; /* We do nothing with "None"-type columns. */
+                    break;
+            }
+
+            if (property->m_valid)
+                trans_props.insert(prop_pair_t(*col_types_it, property));
+            else
+            {
+                parse_errors = loop_err = true;
+                gchar *error_message = g_strdup_printf (_("%s column could not be understood."),
+                                                _(gnc_csv_col_type_strs[*col_types_it]));
+                orig_lines_it->second = error_message;
+
+                g_free (error_message);
```

Comment:
You're right. It would still be faster and better to do
```
std::string error_message {_(gnc_csv_col_type_strs[*col_types_it]);
error_message += " column could not be understood.";
orig_lines_it->second =  std::move(error_message);
```